unleash/Services/ToggleService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ToggleService.swift
+//  Unleash
+//
+//  Copyright © 2019 Silvercar. All rights reserved.
+//
+
+import Foundation
+import PromiseKit
+
+struct ToggleService: ToggleServiceProtocol {
+    private let appName: String
+    private let instanceId: String
+    
+    init(appName: String, instanceId: String) {
+        self.appName = appName
+        self.instanceId = instanceId
+    }
+}
+
+protocol ToggleServiceProtocol {
+    func fetchToggles(url: URL) -> Promise<Features>
+}
+
+extension ToggleService {
+    func fetchToggles(url: URL) -> Promise<Features> {
+        let togglesUrl = url.appendingPathComponent("/client/features")
+        return firstly {
+            URLSession.shared.dataTask(.promise, with: try makeUrlRequest(url: togglesUrl)).validate()
+            }.map {
+                return try JSONDecoder().decode(Features.self, from: $0.data)
```

Comment:
The register service uses `try JSONSerialization.jsonObject(with: $0.data)`, is there a difference using the two?

Reply:
I believe this is how you get it to map to a `Codeble` object.